pkg/placement/selectors.go

```diff
@@ -72,3 +77,97 @@ func GetKeys(m map[string]string) []string {
 	}
 	return keys
 }
+
+// when an object is updated, we iterate over all placements and update placementToResolutionMap that
+// are affected by the update. Every affected placement decision is then queued to by synced.
+// TODO: when PlacementDecision API supports singleton statuses, this function should be updated.
+func (c *Controller) updateDecisions(obj runtime.Object) error {
+	placements, err := c.listPlacements()
+	if err != nil {
+		return err
+	}
+
+	objMR := obj.(mrObject)
+
+	changedResolutions := map[string]interface{}{}
+
+	for _, item := range placements {
+		placement, err := runtimeObjectToPlacement(item)
+		if err != nil {
+			return err
+		}
+
+		matchedSome := c.testObject(objMR, placement.Spec.Downsync)
+		if !matchedSome {
+			// if previously selected, remove
+			// TODO: optimize
+			if resolutionUpdated := c.placementResolver.RemoveObject(placement.GetName(), obj); resolutionUpdated {
+				changedResolutions[placement.GetName()] = nil
+			}
+			continue
+		}
+
+		// obj is selected by placement, update the placement decision resolver
+		resolutionUpdated, err := c.placementResolver.UpdateObjects(placement.GetName(), obj)
+		if err != nil {
+			return fmt.Errorf("failed to update placementToResolutionMap for object %v: %v",
+				util.GenerateObjectInfoString(obj), err)
+		}
+
+		if resolutionUpdated {
+			changedResolutions[placement.GetName()] = nil
+		}
+	}
+
+	// if no decision changed, we can return
+	if len(changedResolutions) == 0 {
+		return nil
+	}
+
+	// get all placement placementToResolutionMap
+	placementDecisions, err := c.listPlacementDecisions()
+	if err != nil {
+		return fmt.Errorf("failed to update placementToResolutionMap: %v", err)
+	}
+
+	handledDecisions := map[string]interface{}{}
+
+	// queue all placement placementToResolutionMap that exist are affected by the update
+	for _, pdObj := range placementDecisions {
+		name := pdObj.(mrObject).GetName()
+		if _, found := changedResolutions[name]; found {
+			// enqueue the placement decision for syncing
+			c.enqueueObject(pdObj, true)
+			// mark as handled
+			handledDecisions[name] = nil
+
+			c.logger.Info("enqueued placement decision", "name", name)
+		}
+	}
+
+	// create all missing placement placementToResolutionMap by enqueuing them as new objects
+	// at a later stage in the pipeline a worker will realize that the object does not exist
+	// and will create it
+	for name, _ := range changedResolutions {
+		if _, found := handledDecisions[name]; found {
```

Comment:
FYI, https://github.com/kubernetes/apimachinery/blob/v0.28.2/pkg/util/sets/set.go#L122 is a thing.

Reply:
A slice of strings now suffices since placement-decisions can be queued by name, with no need to distinguish between existing and none-existing ones.